test/src/main/java/org/apache/accumulo/test/ShellServerIT.java

```diff
@@ -2687,13 +2705,15 @@ public void testCreateTableWithBinarySplitsFile1()
     try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
       splitsFile = System.getProperty("user.dir") + "/target/splitFile";
       generateSplitsFile(splitsFile, 200, 12, true, true, true, false, false);
-      SortedSet<Text> expectedSplits = readSplitsFromFile(splitsFile, false);
+      SortedSet<Text> expectedSplits = readSplitsFromFile(splitsFile);
       final String tableName = name.getMethodName() + "_table";
       ts.exec("createtable " + tableName + " -sf " + splitsFile, true);
       Collection<Text> createdSplits = client.tableOperations().listSplits(tableName);
       assertEquals(expectedSplits, new TreeSet<>(createdSplits));
     } finally {
-      Files.delete(Paths.get(splitsFile));
+      if (Objects.nonNull(splitsFile)) {
```

Comment:
`Objects.nonNull` exists for use as a lambda when a Predicate is required (`stream.filter(Objects::nonNull)`, for example). For plain old if statements, it's much more clear to just write:

```suggestion
      if (splitsFile != null) {
```

However, there's no reason for it to ever be null or for us to check for it being null. We can just assign it to `System.getProperty("user.dir") + "/target/splitFile";` in the first place. There's no reason that needs to be done inside the try block.

Reply:
I'll change these - the "reason" is the code analysis flags these as possibility returning null and didn't want to add suppression(s)

Reply:
Right, but they are only possibly returning null because null is assigned to the variable first. However, that's an unnecessary assignment. The assignment inside each try block can be moved to just before the try block, so they are immediately assigned to the value from reading the system property, and never assign null in the first place. Then, they can never be null, and you don't need the checks to see if they are null in the finally blocks.